### PR TITLE
Calendar interface

### PR DESCRIPTION
**Pull request by zhao19991111**
- Implemented a Calendar interface to select availability
- Implemented the interface in the search bar and the tutor list

### CODE REVIEW

File: frontend/src/components/SeachBar/index.jsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            </Frame>
+          </Frame>
+        }
+      />
+      <FilterCard
+        name="Availability"
+        height={100}
+        cancelFunc={() => {
+          setSchedule([
+            [0, 0, 0, 0, 0, 0, 0],
```
Comment (sammyli0106) [inline frontend/src/components/SeachBar/index.jsx:123]: Is this like resetting the schedule?

File: frontend/src/components/TutorList/index.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+              padding: 5,
+              backgroundColor: themeColors.darkgray,
+            }}
+            onClick={() => {}}
+          />
+        ))}
+      </Frame>
+      <Calendar
+        data={[
+          [0, 0, 0, 1, 1, 1, 1],
```
Comment (sammyli0106) [inline frontend/src/components/TutorList/index.jsx:40]: This is just my curiosity. Is calendar the same as schedule? How do we store and retrieve the user availability?

File: frontend/src/components/Calendar/index.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          onContextMenu={(e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            if (editable) {
+              changeByte(i, j, -1);
+            }
+          }}
+          level={byte}
+        >
+          {String(9 + 2 * j) + "-" + String(9 + 2 * j + 2)}
```
Comment (sammyli0106) [inline frontend/src/components/Calendar/index.jsx:71]: I am not following this. What is the purpose of this? Is that the 2d array you described before?

**Review by sammyli0106 — COMMENTED** [review]
Looks good to me, except just some questions to ask

File: frontend/src/components/SeachBar/index.jsx (hunk 1 above)
Comment (zhao19991111) [reply frontend/src/components/SeachBar/index.jsx:123]: Yeah, it is

File: frontend/src/components/TutorList/index.jsx (hunk 2 above)
Comment (zhao19991111) [reply frontend/src/components/TutorList/index.jsx:40]: Yeah, it's implemented as the same thing. I think the backend would store it as an array of bytes.

File: frontend/src/components/Calendar/index.jsx (hunk 3 above)
Comment (zhao19991111) [reply frontend/src/components/Calendar/index.jsx:71]: It's for the 2 hour range. I will flatten the array to 1D.

**Review by GeraldDT5 — APPROVED** [review]
Sammy's questions look answered. Everything else I saw was good

**Pull request merged**